### PR TITLE
Feature target metadata ui

### PR DESCRIPTION
**Pull request by StefanKlt**
Make target metadata editable via management UI

### CODE REVIEW

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/distributions/dstable/DsMetadataPopupLayout.java [hunk 1]
```diff
@@ -27,7 +27,7 @@
 /**
  * Pop up layout to display distribution metadata.
  */
-public class DsMetadataPopupLayout extends AbstractMetadataPopupLayout<DistributionSet, MetaData> {
+public class DsMetadataPopupLayout extends AbstractMetadataPopupLayoutVersioned<DistributionSet, MetaData> {
```
Comment (StefanKlt) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/distributions/dstable/DsMetadataPopupLayout.java]: set UID to default

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/targettable/TargetDetails.java [hunk 2]
```diff
@@ -326,23 +337,27 @@ protected String getDetailsHeaderCaptionId() {
 
     @Override
     protected boolean isMetadataIconToBeDisplayed() {
-        return false;
+        return true;
```
Comment (StefanKlt) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/targettable/TargetDetails.java]: check if method is needed

**Comment by schabdo** [comment]
@hawkbit-bot verify please

**Comment by schabdo** [comment]
Closes #737

File: hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaTargetManagement.java [hunk 3]
```diff
@@ -182,10 +183,16 @@ public long count() {
 
         final JpaTarget updatedTarget = touch(target);
 
-        return Collections.unmodifiableList(md.stream()
+        final List<TargetMetadata> createdMetadata = Collections.unmodifiableList(md.stream()
                 .map(meta -> targetMetadataRepository
                         .save(new JpaTargetMetadata(meta.getKey(), meta.getValue(), updatedTarget)))
                 .collect(Collectors.toList()));
+
+        // target update event is set to ignore "lastModifiedAt" field so it is
```
Comment (bogdan-bondar) [inline hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaTargetManagement.java]: I would reformulate it as: TargetUpdatedEvent is not sent within the touch() method due to the "lastModifiedAt" field being ignored in JpaTarget

File: hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaTargetManagement.java [hunk 4]
```diff
@@ -227,8 +234,12 @@ public TargetMetadata updateMetaData(final String controllerId, final MetaData m
         toUpdate.setValue(md.getValue());
         // touch it to update the lock revision because we are modifying the
         // target indirectly
-        touch(controllerId);
-        return targetMetadataRepository.save(toUpdate);
+        final JpaTarget target = touch(controllerId);
+        final JpaTargetMetadata matadata = targetMetadataRepository.save(toUpdate);
```
Comment (bogdan-bondar) [inline hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaTargetManagement.java]: typo: better "updatedMetadata"

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/detailslayout/TargetMetadataDetailsLayout.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+    private Long selectedTargetId;
+
+    /**
+     * Initialize the layout.
+     * 
+     * @param i18n
+     *            the i18n service
+     * @param targetManagement
+     *            the target set management service
```
Comment (bogdan-bondar) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/detailslayout/TargetMetadataDetailsLayout.java]: typo: remove "set"

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/targettable/TargetMetadataPopupLayout.java [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+    @Override
+    protected boolean checkForDuplicate(final Target entity, final String value) {
+        return targetManagement.getMetaDataByControllerId(entity.getControllerId(), value).isPresent();
+    }
+
+    @Override
+    protected MetaData createMetadata(final Target entity, final String key, final String value) {
+        final TargetMetadata metaData = targetManagement
+                .createMetaData(entity.getControllerId(), Arrays.asList(entityFactory.generateDsMetadata(key, value)))
```
Comment (bogdan-bondar) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/targettable/TargetMetadataPopupLayout.java]: entiryFactory.generateTargetMetadata

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/targettable/TargetMetadataPopupLayout.java [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+                .get(0);
+        setSelectedEntity(metaData.getTarget());
+        return metaData;
+    }
+
+    @Override
+    protected MetaData updateMetadata(final Target entity, final String key, final String value) {
+        final TargetMetadata metaData = targetManagement.updateMetaData(entity.getControllerId(),
+                entityFactory.generateDsMetadata(key, value));
```
Comment (bogdan-bondar) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/targettable/TargetMetadataPopupLayout.java]: generateTargetMetadata

**Comment by schabdo** [comment]
@hawkbit-bot verify please

**Pull request merged**